Cargo.toml

```diff
@@ -73,8 +74,8 @@ const_format = "0.2"
 ctor = "0.6"
 criterion = { version = "0.7", features = ["html_reports", "async_tokio"] }
 derive_builder = "0.20"
-diesel = { version = "2.3", default-features = false }
-diesel_migrations = { version = "2.2", default-features = false }
+diesel = { git = "https://github.com/ephemerahq/diesel", branch = "codabrink/writable-deserialize", default-features = false }
```

Comment:
we should put this in the [patch] section

Reply:
Also, is there a plan for upstreaming? I think Weiznich would be open to this change, it seems harmless & as long as it is well-documented, I don't see it causing any issues. it would be nice to not worry about maintaining our own fork in the future, esp if we eventually aim to release libxmtp on crates.io